gramps/gen/filters/rules/person/_hascommonancestorwith.py

```diff
@@ -96,17 +106,22 @@ def add_ancs(self, db, person):
     def reset(self):
         self.ancestor_cache = {}
 
-    def has_common_ancestor(self, other):
+    def has_common_ancestor(self, other: Person):
         for handle in self.with_people:
-            if (handle in self.ancestor_cache and self.ancestor_cache[handle]) & (
+            # Requires bitwise and for some reason:
```

Comment:
```suggestion
```
[I can't unresolve my previous comment so adding a new one]
This is actually doing intersection between two dicts

Reply:
It might be clearer to use `intersection()` instead of `&`

Reply:
Do you want to make a PR on this PR? You can do that here in the UI. Just click on "Edit" of that file, make your changes, and select "branch" if you can.

Reply:
Oh, I understand now! I can fix that.

Reply:
> Do you want to make a PR on this PR? You can do that here in the UI. Just click on "Edit" of that file, make your changes, and select "branch" if you can.

I don't have write access to the repo so I'm not sure if I'll be able to do this

Reply:
No problem, I got this one.